src/python/cli_new/lib/cli/config.py

```diff
@@ -119,6 +120,92 @@ def master(self):
 
         return master
 
+    def principal(self):
+        """
+        Return the principal in the configuration file
+        """
+        return self.data["master"].get("principal")
+
+    def secret(self):
+        """
+        Return the secret in the configuration file
+        """
+        return self.data["master"].get("secret")
+
+    def agent_ssl(self):
+        """
+        Return if the agent support ssl
+        """
+        if "agent" in self.data:
+            agent_ssl = self.data["agent"].get("ssl")
+            if agent_ssl is None:
+                return None
+            if not isinstance(agent_ssl, bool):
+                raise CLIException("The 'agent->ssl' field"
+                                   " must be True/False")
+
+            return agent_ssl
+
+        return None
+
+    def agent_ssl_verify(self):
+        """
+        Return if the ssl certificate should be verified
+        """
+        if "agent" in self.data:
+            ssl_verify = self.data["agent"].get("ssl_verify")
+            if ssl_verify is None:
+                return None
+            if not isinstance(ssl_verify, bool):
+                raise CLIException("The 'agent->ssl_verify' field"
+                                   " must be True/False")
+
+            return ssl_verify
+
+        return None
+
+    def agent_timeout(self):
+        """
+        Return the connection timeout of the agent
+        """
+        if "agent" in self.data:
+            timeout = self.data["agent"].get("timeout")
+            if timeout is None:
+                return 5
```

Comment:
I think you can do instead:
```
            timeout = self.data["agent"].get("timeout", 5)
```

Also, maybe better than hard-coding the default could be moved to the function prototype?
```
    def agent_timeout(self, default=5):
```

Reply:
I like the idea with the default value in the prototype. I will use it also for the agent_ssl and agent_ssl_verify. That will also simplify that functions. :+1:  thanks Charles.